kernel-open/nvidia-uvm/uvm_kvmalloc.c

```diff
@@ -257,7 +257,7 @@ static void *alloc_internal(size_t size, bool zero_memory)
     // Make sure that (sizeof(hdr) + size) is what it should be
     BUILD_BUG_ON(sizeof(uvm_vmalloc_hdr_t) != offsetof(uvm_vmalloc_hdr_t *, ptr));
 
-    assert(size <= (1 << 16));
+    assert(size <= (1 << 32));
```

Comment:
The (1 << 16) limit is there because the kernel heap used for these allocations supports a malloc-style interface (where you don't need to specify a size value when freeing memory) only for allocations sizes up to (1  << 16) (see the MAX_MCACHE_ORDER constant in the Nanos source at src/config.h). So it cannot be changed to a larger value.
If we want to remove this limit, we need to be able to use the vmalloc API when allocations larger that (1 << MAX_MCACHE_ORDER) are requested. This would entail:
- changing the UVM_KMALLOC_THRESHOLD definition in kernel_open/nvidia_uvm/uvm_kvmalloc.h from infinity to (1 << MAX_MCACHE_ORDER)
- changing the is_vmalloc_addr() definition in kernel_open/common/inc/nv_nanos.h from false to `(objcache_from_object(u64_from_pointer(p), PAGESIZE_2M) == INVALID_ADDRESS)`
- changing the vfree() definition in kernel_open/common/inc/nv_nanos.h from kfree to NV_KFREE; this means the macro will take an additional size parameter, for which the macro invocations can use the `alloc_size` value of the corresponding uvm_vmalloc_hdr_t struct (all vmalloc allocations use this header)